### PR TITLE
chore(jupyter): Rename JupyterHub to Jupyter in all resources

### PR DESCRIPTION
**Pull request by dgutride**
## Description

Rename any JupyterHub references to Jupyter through all resources (except for any code specific names like services or user names)

Closes: [RHOAIENG-4318](https://issues.redhat.com//browse/RHOAIENG-4318)

## How Has This Been Tested?

Applied all resources using kustomize/oc apply with operator off and validated the name was correct after cache refresh.

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

File: manifests/overlays/apps/apps-addon/starburst/starburst-galaxy-quickstart.yaml [hunk 1]
```diff
@@ -57,22 +57,22 @@ spec:
...
-        2. Find the **JupyterHub** card.
-        3. Click **Launch** on the JupyterHub card to access the JupyterHub **Start a notebook server** page.  A new browser tab will open displaying the **Start a notebook server** page.
+        2. Find the **Jupyter** card.
+        3. Click **Launch** on the Jupyter card to access the Jupyter **Start a notebook server** page.  A new browser tab will open displaying the **Start a notebook server** page.
         4. Select the **Standard Data Science** notebook image and the **Default** container size.
         5. Add the environment variables for **TRINO\_USERNAME**, **TRINO\_PASSWORD**, **TRINO\_HOSTNAME**, and **TRINO\_PORT** using the connection details you obtained in the previous task.
-        6. Click on **Start server** to launch JupyterHub.
+        6. Click on **Start server** to launch Jupyter.
 
         If you have not previously authorized the jupyterhub-hub service account to access your account, the **Authorize Access** page appears prompting you to provide authorization. Inspect the permissions selected by default, and click the **Allow selected permissions** button.
```
Comment (dpanshug) [inline manifests/overlays/apps/apps-addon/starburst/starburst-galaxy-quickstart.yaml:70]: `jupyterhub-hub` is this supposed to be this way?

Comment (dpanshug) [inline manifests/overlays/apps/apps-addon/starburst/starburst-galaxy-quickstart.yaml:70]: `jupyterhub-hub` is this supposed to be this way?

File: manifests/overlays/apps/apps-onprem/starburst-enterprise/query-starburstenterprise-quickstart.yaml [hunk 2]
```diff
@@ -43,40 +43,40 @@ spec:
...
-        2. Find the **JupyterHub** card.
-        3. Click **Launch** on the JupyterHub card to access the JupyterHub **Start a notebook server** page.  A new browser tab will open displaying the **Start a notebook server** page.
+        2. Find the **Jupyter** card.
+        3. Click **Launch** on the Jupyter card to access the Jupyter **Start a notebook server** page.  A new browser tab will open displaying the **Start a notebook server** page.
         4. Select the **Standard Data Science** notebook image and the **Default** container size.
         5. Add the environment variables for **TRINO\_USERNAME**, **TRINO\_PASSWORD**, **TRINO\_HOSTNAME**, and **TRINO\_PORT** using the connection details you obtained in the previous task.
-        6. Click on **Start server** to launch JupyterHub.
+        6. Click on **Start server** to launch Jupyter.
 
         If you have not previously authorized the jupyterhub-hub service account to access your account, the **Authorize Access** page appears prompting you to provide authorization. Inspect the permissions selected by default, and click the **Allow selected permissions** button.
```
Comment (dpanshug) [inline manifests/overlays/apps/apps-onprem/starburst-enterprise/query-starburstenterprise-quickstart.yaml:74]: same here with `jupyterhub-hub`

**Review by dpanshug — COMMENTED** [review]
Looks good, just a few questions

File: manifests/overlays/apps/apps-onprem/starburst-enterprise/query-starburstenterprise-quickstart.yaml (hunk 2 above)
Comment (dgutride) [reply manifests/overlays/apps/apps-onprem/starburst-enterprise/query-starburstenterprise-quickstart.yaml:74]: Those are correct right now - starburst docs were submitted by an external contributor and until we get some feedback from them, I will leave them as is.

**Review by dpanshug — APPROVED** [review]
/lgtm

**Review by harshad16 — COMMENTED** [review]
lgtm
based on the simple browse through documentations.

/lgtm

File: manifests/overlays/apps/apps-addon/nvidia/gpu-quickstart.yaml [hunk 3]
```diff
@@ -12,12 +12,12 @@ spec:
   description: Install GPU add-on
   prerequisites: [Your cluster has at least one GPU node provisioned.]
   introduction: |-
-    This quick start shows you how to install the GPU add-on and verify that the JupyterHub spawner detects the available GPUs.
+    This quick start shows you how to install the GPU add-on and verify that the Jupyter spawner detects the available GPUs.
```
Comment (bredamc) [inline manifests/overlays/apps/apps-addon/nvidia/gpu-quickstart.yaml:15]:
```suggestion
    This quick start shows you how to install the GPU add-on and verify that the Jupyter Spawner detects the available GPUs.
```

Initcap for Spawner (not spawner)?

File: manifests/overlays/apps/apps-addon/starburst/starburst-galaxy-quickstart.yaml (hunk 1 above)
Comment (bredamc) [inline manifests/overlays/apps/apps-addon/starburst/starburst-galaxy-quickstart.yaml:70]:
```suggestion
        If you have not previously authorized the `jupyter-hub` service account to access your account, the **Authorize Access** page appears prompting you to provide authorization. Inspect the permissions selected by default, and click the **Allow selected permissions** button.
```

Two suggestions:
1. Remove "hub" from the service account name?
2. Enclose the service name in backticks for consistency with OpenShift doc convention?

File: manifests/overlays/apps/apps-onprem/starburst-enterprise/query-starburstenterprise-quickstart.yaml (hunk 2 above)
Comment (bredamc) [inline manifests/overlays/apps/apps-onprem/starburst-enterprise/query-starburstenterprise-quickstart.yaml:74]: Same comments as for previous file:
1.     Remove "hub" from the service account name?
2.     Enclose the service name in backticks for consistency with OpenShift doc convention?

File: manifests/overlays/apps/base/aikit/oneapi-aikit-notebook-quickstart.yaml [hunk 4]
```diff
@@ -76,44 +76,44 @@ spec:
...
+        2. (Optional) You can proceed with this step and the previous step with any of the either samples in the correct AI Kit Jupyter Kernel environment to learn more about other AI Kit optimizations. Use the `Intel SKLearn, XGBoost, & Modin` Jupyter Kernel Environment for the `IntelPython_XGBoost_daal4pyPrediction.ipynb` notebook example, and the `Intel TensorFlow & Quantization` Jupyter Kernel Environment for the `lpot_sample_tensorflow.ipynb` notebook example.
       review:
         instructions: |-
           #### Verify that the sample notebook ran successfully
           Did all cells of the notebook run successfully and printed out the statement "[CODE\_SAMPLE\_COMPLETED\_SUCCESSFULLY]" at the end?
         failedTaskHelp:
           This task is not verified yet. Try the task again.
       summary:
-        success: You have run the example notebook in the correct AI Kit JupyterHub Kernel environment.
+        success: You have run the example notebook in the correct AI Kit Jupyter Kernel environment.
```
Comment (bredamc) [inline manifests/overlays/apps/base/aikit/oneapi-aikit-notebook-quickstart.yaml:101]:
```suggestion
        success: You have run the example notebook in the correct AI Kit Jupyter Kernel Environment.
```

Initcap for Jupyter Kernel Environment (not environment)? If so, search and replace globally.

**Review by bredamc — COMMENTED** [review]
Just a few minor comments, feel free to ignore them.

Also wondering whether you want to update the occurrences in the following files:

- [backend/src/utils/route-security.ts](https://github.com/opendatahub-io/odh-dashboard/blob/b1673c004561f2212cb54c168e4ab2322a3a024a/backend/src/utils/route-security.ts#L116)
- [backend/src/utils/notebookUtils.ts](https://github.com/opendatahub-io/odh-dashboard/blob/b1673c004561f2212cb54c168e4ab2322a3a024a/backend/src/utils/notebookUtils.ts#L39)
- [sts/overlays/rhoai/anaconda-ce-validator-cron.yaml](https://github.com/opendatahub-io/odh-dashboard/blob/b1673c004561f2212cb54c168e4ab2322a3a024a/manifests/overlays/rhoai/anaconda-ce-validator-cron.yaml#L39)
- [frontend/src/utilities/notebookControllerUtils.ts](https://github.com/opendatahub-io/odh-dashboard/blob/b1673c004561f2212cb54c168e4ab2322a3a024a/frontend/src/utilities/notebookControllerUtils.ts#L46)

File: manifests/overlays/apps/apps-addon/starburst/starburst-galaxy-quickstart.yaml (hunk 1 above)
Comment (dgutride) [reply manifests/overlays/apps/apps-addon/starburst/starburst-galaxy-quickstart.yaml:70]: The service name is provided by a third party and is not referring to an actual proper name of something - we need to leave it that way until we hear from the Starburst team.  We should not make this change now.

File: manifests/overlays/apps/apps-onprem/starburst-enterprise/query-starburstenterprise-quickstart.yaml (hunk 2 above)
Comment (dgutride) [reply manifests/overlays/apps/apps-onprem/starburst-enterprise/query-starburstenterprise-quickstart.yaml:74]: See comment above - this is provided by a third party.

**Comment by dgutride** [comment]
@bredamc  - thanks for the review - the other changes suggested in files with service names or other names are not part of this (and would be incorrect because they would break functionality) - this is only for display names in the resource files.

**Review by andrewballantyne — APPROVED** [review]

**Pull request merged**